http-transport-binding.md

```diff
@@ -0,0 +1,363 @@
+is mapped to a separate HTTP header. The `extensions` attribute itself is
+not mapped to a header.
+
+The naming convention for the `extensions` header mapping of attributes is:
+
+    * Each map entry name MUST be prefixed with "CE-X-"
+    * Each map entry name's first character MUST be capitalized
+
+Examples:
+
+    * `example` maps to `CE-X-Example`
+    * `testExtension` maps to `CE-X-TestExtension`
+
+##### 3.1.3.2 HTTP Header Values
+
+The value for each HTTP header is constructed from the respective attribute's
+[JSON value][JSON-value] representation, compliant with the [JSON event
+format][JSON-format] specification.
+
+Some CloudEvents metadata attributes can contain arbitrary UTF-8 string
+content, and per [RFC7230 Section 3][RFC7230-Section-3], HTTP headers MUST only
+use printable characters from the US-ASCII character set, and are terminated by
+a CRLF sequence.
+
+Therefore, and analog to the encoding rules for Universal character set host
+names in URIs [RFC3986 3.2.2][RFC3986], the JSON value MUST be encoded as
+follows:
+
+Non-printable ASCII characters and non-ASCII characters MUST first be encoded
+according to UTF-8, and then each octet of the corresponding UTF-8 sequence
+MUST be percent-encoded to be represented as HTTP header characters, in
+compliance with [RFC7230, sections 3, 3.2, 3.2.6][RFC7230-Section-3].
+
+JSON objects and arrays are NOT surrounded with single or double quotes.
+
+#### 3.1.4 Examples
+
+This example shows the *binary* mode mapping of an event with an HTTP POST
+request:
+
+``` text
+POST /someresource HTTP/1.1
+Host: webhook.example.com
+CE-CloudEventsVersion: "0.1"
+CE-EventType: "com.example.someevent"
+CE-EventTime: "2018-04-05T03:56:24Z"
+CE-EventID: "1234-1234-1234"
+CE-Source: "/mycontext/subcontext"
+    .... further attributes ...
+Content-Type: application/json; charset=utf-8
+Content-Length: nnnn
+
+{
+    ... application data ...
+}
+```
+
+This example shows a response containing an event:
+
+``` text
+HTTP/1.1 200 OK
+CE-CloudEventsVersion: "0.1"
+CE-EventType: "com.example.someevent"
+CE-EventTime: "2018-04-05T03:56:24Z"
+CE-EventID: "1234-1234-1234"
+CE-Source: "/mycontext/subcontext"
+    .... further attributes ...
+Content-Type: application/json; charset=utf-8
+Content-Length: nnnn
+
+{
+    ... application data ...
+}
+```
+
+### 3.2. Structured Content Mode
+
+The *structured* content mode keeps event metadata and data together in the
+payload, allowing simple forwarding of the same event across multiple routing
+hops, and across multiple transports.
+
+#### 3.2.1. HTTP Content-Type
+
+The [HTTP `Content-Type`][Content-Type] header is set to the media type of an
+[event format](#14-event-formats).
+
+Example for the [JSON format][JSON-format]:
+
+``` text
+Content-Type: application/cloudevents+json; charset=UTF-8
+```
+
+#### 3.2.2. Event Data Encoding
+
+The chosen [event format](#14-event-formats) defines how all attributes,
```

Comment:
I vehemently agree with this statement and wish it could be hoisted up even higher (e.g. this is not just a nuance of HTTP, but that CloudEvents will not support mixed-encoding).

This is the meat behind my argument that "content-type" is a requirement of every transport which we define in this spec, but is not itself a feature of a CloudEvent.

Reply:
Content-Type is required in the event, because it designates the event payload and not the event.

Reply:
It really is about encoding.  Maybe it should be a separate part of the CloudEvent spec, since it is about how the event is formatted, rather than the semantic of the event itself.  (This was why I originally thought it didn't belong in the spec, but can see that people really want to put *some* encoding semantics in the spec itself.  I'm ok with leaving this interpretation for now, and then seeing how it plays out in implementations.)